### DISCUSSION TITLE
Webtop LSIO : Why?

### DISCUSSION THREAD

**Discussion by UlrichThiess** [alexbelgium/hassio-addons#1555] [Q&A] [answered]
The Linux provided with the KDE interface invites you to install software and try things out. Admittedly, as a container on a home assistant system it only makes limited sense, but if everything is reset after every restart, the purpose is not clear to me at all.

Can this container be set to read/write? To get an unmanipulated container, you can simply reinstall the add-on. What would be the best way to proceed?

**Comment by alexbelgium** [comment, marked as answer]
Hi, it's not a possibility of homeassistant. However, you can use the option in the add-on options page to install additional apps at startup

Another option is to use my injection script to install apps at boot https://github.com/alexbelgium/hassio-addons/wiki/Add%E2%80%90ons-feature-:-customisation

Good luck!

**Comment by UlrichThiess** [comment]
That's what I was afraid of. Thank you for your answer.

**Comment by UlrichThiess** [comment]
I followed your recommendation
```
➜  pwd
/homeassistant/addons_config/webtop-kde
➜  ls -ahl
total 16K    
drwxr-xr-x    2 root     root        4.0K Aug 29 10:16 .
drwxr-xr-x    3 root     root        4.0K Aug 27 21:15 ..
-rwxr-xr-x    1 root     root         642 Aug 27 21:15 config.yaml
-rwxr-xr-x    1 root     root         482 Aug 29 10:27 webtop-kde.sh
➜  cat webtop-kde.sh 
#!/usr/bin/env bashio
echo "Starting script"
# Installing cron
apt-get update
apt-get install cron -yqq
# Setting cron
echo "echo \"hello\"" > /etc/cron.hourly/hello
chmod 777 /etc/cron.hourly/hello
# Starting cron
service cron start & true
```

#### log
```
Timezone set from Etc/UTC to Europe/Berlin
/etc/cont-init.d/00-local_mounts.sh: executing
/etc/cont-init.d/00-smb_mounts.sh: executing
/etc/cont-init.d/01-config_yaml.sh: executing
Setting permissions for the config.yaml directory

Load environment variables from /config/addons_config/webtop-kde/config.yaml if existing
If accessing the file with filebrowser it should be mapped to /homeassistant/addons_config/webtop-kde/config.yaml
---------------------------------------------------------
Wiki here on how to use : github.com/alexbelgium/hassio-addons/wiki/Add‐ons-feature-:-add-env-variables

... no env variables found, exiting
/etc/cont-init.d/01-custom_script.sh: executing
Execute /homeassistant/addons_config/webtop-kde/webtop-kde.sh if existing
Wiki here : github.com/alexbelgium/hassio-addons/wiki/Add-ons-feature-:-customisation
/etc/cont-init.d/20-folders.sh: executing
[10:27:59] INFO: Setting data location to /share/webtop_kde
usermod: no changes
Creating /share/webtop_kde
[10:28:00] INFO: Setting ownership to 0:0
/etc/cont-init.d/80-configuration.sh: executing
[10:28:00] INFO: Installing additional apps :
... engrampa
/etc/cont-init.d/90-dns_set.sh: executing
[10:28:02] INFO: DNS SERVERS set to 192.168.0.3 
/etc/cont-init.d/90-ingress.sh: executing
 
Starting the upstream container
```

the scrip does not run?
should dns setting start first?

**Reply by alexbelgium** [reply]
I see - this is because this addon is still with the "old" way to manage /config folder (= prior to HomeAssistant implementing a feature similar to mine).

The file to access is /homeassistant/addons_autoscripts/webtop-kde.sh - I see that my code has a typo and does not show the correct path in the log

**Comment by UlrichThiess** [comment]
Thanks again. Now I can move on...
